src/psyclone/psyir/transformations/acc_update_trans.py

```diff
@@ -0,0 +1,332 @@
+                        self._add_updates_to_schedule(child.else_body)
+                elif isinstance(child, Loop):
+                    # A loop on the CPU with code that may be on the GPU.
+                    # The loop start, stop and step values could all
+                    # potentially have been previously written to on the GPU.
+                    self._add_update_directives([child.start_expr,
+                                                 child.stop_expr,
+                                                 child.step_expr])
+                    # Recurse down into the loop body
+                    self._add_updates_to_schedule(child.loop_body)
+
+        # We've reached the end of the list of children - are there any
+        # last nodes that represent computation on the CPU?
+        self._add_update_directives(node_list)
+
+    def _add_update_directives(self, node_list):
+        '''
+        Adds the required update directives before and after the nodes in
+        the supplied list.
+
+        :param sched: the schedule which contains the nodes in node_list.
+        :type sched: :py:class:`psyclone.psyir.nodes.Schedule`
+        :param node_list: the PSyIR nodes representing code executed on the \
+                          CPU rather than the GPU.
+        :type node_list: list of :py:class:`psyclone.psyir.nodes.Node`
+
+        '''
+        if not node_list:
+            return
+
+        inputs, outputs = self._dep_tools.get_in_out_parameters(node_list)
+        inputs, outputs = set(inputs), set(outputs)
+
+        # Workaround for lack of precise access descriptions, e.g. in the
+        # presence of array slices we currently overapproximate dependencies.
+        inputs.update(outputs)
+
+        # Workaround for CodeBlocks
+        # TODO We are probably relying on undefined behaviour here.
+        for node in node_list:
+            if isinstance(node, CodeBlock):
+                for symbol_name in node.get_symbol_names():
+                    inputs.add(Signature(symbol_name))
+                    outputs.add(Signature(symbol_name))
+
+        # Copy any data that is read by this region to the host (resp. device)
+        # if it is on the device (resp. host).
+        IN, OUT = 0, 1
+        for idx, cpu_sig in enumerate((inputs, outputs)):
+            if not cpu_sig:
+                continue
+            if idx == IN:     # inputs
+                node_index = 0
+                node_offset = 0
+                direction = "host"
+                tentative_update_pos = "beg"
+            elif idx == OUT:  # outputs
+                node_index = -1
+                node_offset = 1
+                direction = "device"
+                tentative_update_pos = "end"
+
+            # Since the supplied nodes may be the children of an IfBlock or
+            # Loop, we have to search up to find the ancestor that *is* a
+            # child of a Schedule.
+            child = node_list[node_index]
```

Comment:
You can do `sched = node_list[node_index].ancestor(Schedule)`

Reply:
How would I then get the child of `sched` which is an ancestor of `child`?